MaxMind.Db.Benchmark/Program.cs

```diff
@@ -1,98 +1,239 @@
-﻿#region
-
+﻿using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Running;
+using MaxMind.Db;
 using System;
 using System.Collections.Generic;
-using System.Diagnostics;
-using System.IO;
+using System.Collections.ObjectModel;
+using System.Linq;
 using System.Net;
 
-#endregion
+BenchmarkRunner.Run<CityBenchmark>(new DebugInProcessConfig());
 
-namespace MaxMind.Db.Benchmark
+[MemoryDiagnoser]
+public class CityBenchmark
 {
-    public class Country
+    // A random IP that has city info.
+    private Reader _reader = null!;
+    private IPAddress[] _ipAddresses = [];
+
+    [GlobalSetup]
+    public void GlobalSetup()
     {
-        public string? IsoCode;
+        const string dbPathVarName = "MAXMIND_BENCHMARK_DB";
+        string dbPath = Environment.GetEnvironmentVariable(dbPathVarName) ??
+                        throw new InvalidOperationException($"{dbPathVarName} was not set");
+        _reader = new Reader(dbPath);
 
-        public Country()
+        const string ipAddressesVarName = "MAXMIND_BENCHMARK_IP_ADDRESSES";
+        string ipAddressesStr = Environment.GetEnvironmentVariable(ipAddressesVarName) ?? "";
+        _ipAddresses = ipAddressesStr
+            .Split(',', StringSplitOptions.RemoveEmptyEntries)
```

Comment:
It looks like the framework build still failed for the benchmarks, despite the workflow "passing":

```
Error: D:\a\MaxMind-DB-Reader-dotnet\MaxMind-DB-Reader-dotnet\MaxMind.Db.Benchmark\Program.cs(31,25): error CS1503: Argument 2: cannot convert from 'System.StringSplitOptions' to 'char' [D:\a\MaxMind-DB-Reader-dotnet\MaxMind-DB-Reader-dotnet\MaxMind.Db.Benchmark\MaxMind.Db.Benchmark.csproj::TargetFramework=net481]
```

Reply:
Done. Tested on windows this time.